Tools/WebKitTestRunner/mac/EventSenderProxy.mm

```diff
@@ -950,4 +950,15 @@ static CGMomentumScrollPhase cgMomentumPhaseFromPhase(EventSenderProxy::WheelEve
 
 #endif // ENABLE(MAC_GESTURE_EVENTS)
 
+void EventSenderProxy::waitForPendingMouseEvents()
+{
+    if (NSView *targetView = std::exchange(m_targetView, nullptr)) {
+        __block bool doneProcessingMouseEvents = false;
+        [checked_objc_cast<WKWebView>(targetView) _doAfterProcessingAllPendingMouseEvents:^{
+            doneProcessingMouseEvents = true;
+        }];
+        m_testController->runUntil(doneProcessingMouseEvents, 100_ms);
```

Comment:
I imagine we'll want to wait forever, right?

Reply:
There are rare situations that may cause this callback to not be called, like if the WP exits before the UI process receives a reply that the mouse event was handled.